### ISSUE TITLE
POSSIBLE --bug report -- Not sure of expected behavior to be honest..

### ISSUE THREAD

**Issue by sjmcdowall** [zino-app/graphql-flutter#121]
**Describe the bug**
It appears that the Query -> builder:  part can get called multiple times the 2nd (and subsequent times) a Widget using it is invoked...

**To Reproduce**
1.  Create a simple Stateful component that does a simple Query and generates a list of items (say) 
2.  Put lots of print() debug statements are all the areas of the code to see what's going on :) 
3.  Invoke the component (via say a Drawer and a pushNamed) ... you will note that it appears normal behavior happens the first time after a Reload ... 
4.  Hit the back button and then hit the button to cause the page to load again ..
5.  Note that the Query -> builder part is called twice with data 

This was done with out any Fetch options for cache (so default should have been cache) .. 

**Expected behavior**
I expected the same behavior the 2nd and other times as the first time -- to invoked the builder with the result.data only once, not multiple times (with the same data) 

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: iPhone X simulator 
 - OS: iOS 12 

*** graphql_flutter ***
graphql_flutter:
    dependency: "direct main"
    description:
      name: graphql_flutter
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.0.0-alpha.10"

**Additional context**
None -- just sort of wondering what is going on here... I have a lot of print() debug statements to show what is being invoked (it's not the scopedmodel or general build() for the widget, JUST the Query -> builder) ..  

TIA!

**Comment by HofmannZ** [comment]
@sjmcdowall All Queries are handed as teams in the query manager, so it's possible you receive data twice. On the other side the stream should be closed once the query completes, so I will investigate this behaviour and check if nothing funky goes on in the scheduler.

**Comment by micimize** [comment]
This is expected behavior - the 1st result with data is with results from the cache, then the second is the result from the network.

with `fetchPolicy: FetchPolicy.noCache` there is no "double tap". It would be good to allow the user to more straightforwardly distinguish result types, which I'm going to fold into #153
